src/month_calendar/index.es

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { formatNumber as pad } from 'nebenan-helpers/lib/formatters';
+import { arrayOf } from 'nebenan-helpers/lib/data';
+import {
+  getMonthDetails,
+  getMonthLabel,
+  getMonthString,
+  getToday,
+  isAfter,
+  isBefore,
+  isMonthInYearRange,
+} from './utils';
+
+
+const Controls = ({ label, onNext, onPrevious }) => {
+  const handlePrevious = () => {
+    if (onPrevious) return onPrevious();
+  };
+  const handleNext = () => {
+    if (onNext) return onNext();
+  };
+
+  const previousClasses = clsx('c-datepicker-controls-previous', {
+    'is-disabled': !onPrevious,
+  });
+
+  const nextClasses = clsx('c-datepicker-controls-next', {
+    'is-disabled': !onNext,
+  });
+
+  return (
+    <div className="c-datepicker-controls">
+      <i onClick={handlePrevious} className={previousClasses}>
+        ←
+      </i>
+      <span className="c-datepicker-controls-label">{label}</span>
+      <i onClick={handleNext} className={nextClasses}>
+        →
+      </i>
+    </div>
+  );
+};
+
+const DAYS_COUNT = 7;
+
+const renderLabelsRow = (locale) => {
+  const { firstDay, modules: { calendar: { weekdaysShort } } } = locale;
+  const labels = arrayOf(DAYS_COUNT).map((i) => {
+    const key = i + firstDay > 6 ? 0 : i + firstDay;
+    const className = clsx('c-datepicker-label', {
+      'is-weekend': key === 6 || key === 0,
+    });
+    return <th key={key} className={className}>{weekdaysShort[key]}</th>;
+  });
+  return <tr>{labels}</tr>;
+};
+
+const renderMonth = ({ month, minDate, maxDate, selected, onCellClick }) => {
+  const { days, offset } = getMonthDetails(month);
+
+  const weeks = Math.ceil((days + offset) / DAYS_COUNT);
+  const today = getToday();
+
+  const renderWeek = (week) => {
+    const shift = week * DAYS_COUNT;
+    const isFirstWeek = week === 0;
+
+    const items = arrayOf(DAYS_COUNT).map((day) => {
+      const date = shift + day + 1 - offset;
+
+      const key = `${month}-${pad(date)}`;
```

Comment:
can we put date in state directly? and compare dates with date-fns `isSameDay` util.

Reply:
What would be the improvement? Performance/readability?

Reply:
not sure, this looks more performant to me compared to using date object comparison in dateFns